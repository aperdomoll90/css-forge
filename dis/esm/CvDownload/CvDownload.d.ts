import React from 'react';
import './styles.css';
import '../utils/GlobalStyles.css';
import { CvDownloadPropsType } from './CvDownload.types';
export declare const CvDownload: React.FC<CvDownloadPropsType>;
