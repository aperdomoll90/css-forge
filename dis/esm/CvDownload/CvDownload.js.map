{"version":3,"file":"CvDownload.js","sourceRoot":"","sources":["../../../src/CvDownload/CvDownload.tsx"],"names":[],"mappings":";;;;;;;;;;;;AACA,OAAO,cAAc,CAAA;AACrB,OAAO,2BAA2B,CAAA;AAClC,OAAO,IAAI,MAAM,kBAAkB,CAAA;AACnC,OAAO,EAAE,cAAc,EAAE,MAAM,kBAAkB,CAAA;AAEjD,OAAO,EAAE,kBAAkB,EAAE,MAAM,4BAA4B,CAAA;AAE/D,MAAM,CAAC,IAAM,UAAU,GAAkC,UAAC,EAA2C;QAAzC,KAAK,WAAA,EAAE,WAAW,iBAAA,EAAE,UAAU,gBAAA,EAAE,OAAO,aAAA;IACjG,OAAO,CACL,wBACE,SAAS,EAAC,qBAAqB,EAC/B,KAAK,EAAE,EAAE,SAAS,EAAE,KAAK,CAAC,CAAC,CAAC,KAAK,CAAC,CAAC,CAAC,SAAS,EAAE,eAAe,EAAE,KAAK,CAAC,CAAC,CAAC,kBAAkB,CAAC,KAAK,EAAE,CAAC,EAAE,CAAC,CAAC,CAAC,CAAC,kBAAkB,CAAC,SAAS,EAAE,CAAC,EAAE,CAAC,EAAE,eAAe,EAAE,WAAW,CAAC,CAAC,CAAC,WAAW,CAAC,CAAC,CAAC,OAAO,EAAE,cAAc,EAAE,UAAU,CAAC,CAAC,CAAC,UAAU,CAAC,CAAC,CAAC,OAAO,EAAyB,iBAC7Q,cAAK,GAAG,EAAE,IAAI,EAAE,SAAS,EAAC,kBAAkB,GAAG,EAC9C,cAAc,EACf,qBAAG,IAAI,EAAE,OAAO,EAAE,MAAM,EAAC,QAAQ,EAAC,SAAS,EAAC,0BAA0B,gBACpE,cAAK,SAAS,EAAC,mBAAmB,GAAG,IACnC,EACJ,qBAAG,SAAS,EAAC,mBAAmB,8BAAa,KACzC,CACP,CAAA;AACH,CAAC,CAAA"}