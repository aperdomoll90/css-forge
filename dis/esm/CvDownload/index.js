export { CvDownload } from "./CvDownload";
export * from './CvDownload.types';
//# sourceMappingURL=index.js.map