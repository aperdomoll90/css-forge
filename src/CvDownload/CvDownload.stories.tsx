import { Meta, StoryFn } from '@storybook/react'
import { CvDownload } from './CvDownload'
import StorybookContainer from '../StorybookContainer'
import { CvDownloadPropsType } from './CvDownload.types'
import { dozen } from '../utils/StoryProps'

export default {
  title: 'Buttons',
  component: CvDownload,
  parameters: {
    jest: ['CvDownload.test.tsx'],
  },
  argTypes: {
    size: { options: dozen, control: { type: 'select' } },
    color: { options: ['', '#ff0000', 'rgb(0, 0, 255)', '#1c2942'], control: { type: 'radio' } },
    buttonHover: { options: ['', '#ff0000', 'rgb(0, 0, 255)', '#1c2942'], control: { type: 'radio' } },
    labelColor: { options: ['#fff', '#418440', '#1c2942'], control: { type: 'radio' } },
  },
} as Meta

const CvDownloadTemplate: StoryFn<CvDownloadPropsType> = args => (
  <StorybookContainer>
    <CvDownload {...args} />
  </StorybookContainer>
)

export const CvDownloadComponent = CvDownloadTemplate.bind({})
CvDownloadComponent.args = {
  linkUrl: 'test',
  color: '',
  buttonHover: '',
  labelColor: '',
}
