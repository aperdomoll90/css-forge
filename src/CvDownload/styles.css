.cv-download-wrapper {
  height: var(--size);
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  filter: drop-shadow(3px 5px 2px rgb(0 0 0 / 0.4));
}
.cv-download-page {
  height: 100%;
  fill: #fff;
}

.cv-download-page .page-corner {
  fill: #e7ecec;
}
.cv-download-page .page-line {
  fill: #cdccca;
  stroke: #cdccca;
  stroke-miterlimit: 10;
  stroke-width: 3px;
}
.cv-download-banner {
  width: 120%;
  color: rgb(0, 0, 255);
  position: absolute;
}
.cv-download-banner-path {
  fill: var(--color);
}
.cv-download-banner-underbanner {
  fill: var(--underBanner);
}
.cv-download-banner-letters {
  fill: white;
}
.cv-download-arrow-circle {
  height: 28%;
  width: 35%;
  border-radius: 50%;
  position: absolute;
  bottom: -10%;
  right: -10%;
  display: flex;
  align-items: center;
  justify-content: center;
  background: var(--color);
  box-shadow: -2px -2px 10px 2px rgba(0, 0, 0, 0.406);
  transition: 0.5s ease-in-out;
}

.cv-download-arrow {
  transform: scale(0.45);
  min-height: calc(var(--size) / 2.2);
  min-width: calc(var(--size) / 2.2);
  background: #f5f6f6;
  filter: drop-shadow(-2px -2px 3px rgba(0, 0, 0, 0.481));
  clip-path: polygon(18% 60%, 34% 60%, 34% 0, 68% 0, 68% 60%, 84% 60%, 52% 100%);
  transition: 0.5s ease-in-out;
}

.cv-download-wrapper:hover .cv-download-arrow-circle {
  background: var(--buttonHover);
  transform: rotate(-360deg) scale(3);
  bottom: 30%;
  right: 30%;
  box-shadow: -2px -2px 10px 10px rgba(0, 0, 0, 0.615);
}

.cv-download-label {
  font-size: 100%;
  font-weight: 600;

  color: var(--labelColor);
  position: absolute;
  bottom: -50%;
  letter-spacing: 3px;
  opacity: 0;
  transition: 0.5s ease-in-out;
}

.cv-download-wrapper:hover .cv-download-label {
  bottom: -35%;
  opacity: 1;
}
