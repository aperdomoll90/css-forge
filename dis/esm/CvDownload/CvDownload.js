var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import './styles.css';
import '../utils/GlobalStyles.css';
import page from './media/page.svg';
import { DownloadBanner } from './media/CvBanner';
import { LightenDarkenColor } from '../utils/ColorManipulation';
export var CvDownload = function (_a) {
    var color = _a.color, buttonHover = _a.buttonHover, labelColor = _a.labelColor, linkUrl = _a.linkUrl;
    return (_jsxs("div", __assign({ className: 'cv-download-wrapper', style: { '--color': color ? color : '#71bd55', '--underBanner': color ? LightenDarkenColor(color, -80) : LightenDarkenColor('#71bd55', -80), '--buttonHover': buttonHover ? buttonHover : 'green', '--labelColor': labelColor ? labelColor : 'white' } }, { children: [_jsx("img", { src: page, className: 'cv-download-page' }), DownloadBanner, _jsx("a", __assign({ href: linkUrl, target: '_blank', className: 'cv-download-arrow-circle' }, { children: _jsx("div", { className: 'cv-download-arrow' }) })), _jsx("p", __assign({ className: 'cv-download-label' }, { children: "DOWNLOAD" }))] })));
};
//# sourceMappingURL=CvDownload.js.map