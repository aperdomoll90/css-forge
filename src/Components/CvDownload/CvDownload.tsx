import React from 'react'
import './styles.css'
import page from './media/page.svg'
import banner from './media/CvBanner.svg'

export default function CvDownload() {
  return (
    <div id='cv-download-wrapper'>
      <img src={page} id='cv-download-page' />
      <img src={banner} id='cv-download-banner' />
      <div id='cv-download-arrow-circle'>
        <div id='cv-download-arrow' />
          </div>
          <p id='cv-download-label'>DOWNLOAD</p>
    </div>
  )
}
