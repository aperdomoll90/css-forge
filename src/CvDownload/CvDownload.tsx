import React from 'react'
import './styles.css'
import page from './media/page.svg'
import { DownloadBanner } from './media/CvBanner'
import { CvDownloadPropsType } from './CvDownload.types'
import { LightenDarkenColor } from '../utils/ColorManipulation'

export const CvDownload: React.FC<CvDownloadPropsType> = ({ color, buttonHover, labelColor, linkUrl }) => {
  return (
    <div
      className='cv-download-wrapper'
      style={{ '--color': color ? color : '#71bd55', '--underBanner': color ? LightenDarkenColor(color, -80) : LightenDarkenColor('#71bd55', -80), '--buttonHover': buttonHover ? buttonHover : 'green', '--labelColor': labelColor ? labelColor : 'white' } as React.CSSProperties}>
      <img src={page} className='cv-download-page' />
      {DownloadBanner}
      <a href={linkUrl} target='_blank' className='cv-download-arrow-circle'>
        <div className='cv-download-arrow' />
      </a>
      <p className='cv-download-label'>DOWNLOAD</p>
    </div>
  )
}
