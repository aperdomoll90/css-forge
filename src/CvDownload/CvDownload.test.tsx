import { render } from '@testing-library/react'
import { CvDownload } from './CvDownload'
import { CvDownloadPropsType } from './CvDownload.types'

describe('CvDownload component', () => {
  const prop: CvDownloadPropsType = {
    linkUrl: 'test',
    color: '#fff',
    buttonHover: 'red',
    labelColor: 'green',
  }
  const renderComponent = () => render(<CvDownload {...prop} />)
  it('should match snapshots', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
