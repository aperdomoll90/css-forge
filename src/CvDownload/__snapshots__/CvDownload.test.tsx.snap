// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvDownload component should match snapshots 1`] = `
<div>
  <div
    class="cv-download-wrapper"
    style="--size: 10rem; --color: #fff; --underBanner: #af; --buttonHover: red; --labelColor: green;"
  >
    <svg
      class="cv-download-page"
      viewBox="0 0 116.89 146.32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <style />
      </defs>
      <g
        data-name="Layer 2"
        id="Layer_2"
      >
        <g
          data-name="Layer 1"
          id="Layer_1-2"
        >
          <polygon
            class="page-paper"
            points="0 146.32 116.16 146.28 116.89 21.55 95.34 0.15 0.07 0 0 146.32"
          />
          <path
            class="page-corner"
            d="M95.05.65V21.37a.09.09,0,0,0,.09.09l21,.22a.08.08,0,0,0,.06-.15L95.21.58A.09.09,0,0,0,95.05.65Z"
          />
          <line
            class="page-line"
            x1="13.44"
            x2="102.62"
            y1="27.4"
            y2="27.78"
          />
          <line
            class="page-line"
            x1="13.34"
            x2="102.52"
            y1="35.08"
            y2="35.47"
          />
          <line
            class="page-line"
            x1="14.02"
            x2="103.2"
            y1="42.6"
            y2="42.98"
          />
          <line
            class="page-line"
            x1="14.01"
            x2="103.19"
            y1="49.89"
            y2="50.27"
          />
          <line
            class="page-line"
            x1="14.09"
            x2="103.27"
            y1="65.44"
            y2="65.82"
          />
          <line
            class="page-line"
            x1="14.2"
            x2="103.37"
            y1="72.78"
            y2="73.16"
          />
          <line
            class="page-line"
            x1="14.18"
            x2="103.35"
            y1="80.06"
            y2="80.45"
          />
          <line
            class="page-line"
            x1="13.85"
            x2="103.03"
            y1="87.87"
            y2="88.25"
          />
          <line
            class="page-line"
            x1="14.05"
            x2="103.23"
            y1="95.54"
            y2="95.92"
          />
          <line
            class="page-line"
            x1="14.2"
            x2="103.37"
            y1="102.99"
            y2="103.37"
          />
          <line
            class="page-line"
            x1="14.17"
            x2="103.34"
            y1="110.49"
            y2="110.87"
          />
          <line
            class="page-line"
            x1="14.08"
            x2="103.26"
            y1="118.03"
            y2="118.41"
          />
          <line
            class="page-line"
            x1="14.31"
            x2="103.49"
            y1="125.66"
            y2="126.04"
          />
          <line
            class="page-line"
            x1="14.05"
            x2="103.23"
            y1="58.03"
            y2="58.41"
          />
          <line
            class="page-line"
            x1="14.39"
            x2="103.57"
            y1="133.12"
            y2="133.5"
          />
        </g>
      </g>
    </svg>
    <svg
      class="cv-download-banner"
      viewBox="0 0 136.74 53.94"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <style />
      </defs>
      <g
        data-name="Layer 2"
        id="Layer_2"
      >
        <g
          data-name="Layer 1"
          id="Layer_1-2"
        >
          <path
            class="cv-download-banner-underbanner"
            d="M10.48,53V37.54L.63,38l-.1,9.12a4.6,4.6,0,0,0,.31,2.14,4.92,4.92,0,0,0,.7,1.22A8.11,8.11,0,0,0,5.18,53,8.12,8.12,0,0,0,10.48,53Z"
          />
          <path
            class="cv-download-banner-underbanner"
            d="M126.32,53.46V38.05l9.85.41.1,9.13a4.61,4.61,0,0,1-.31,2.14,4.83,4.83,0,0,1-.7,1.21,7.95,7.95,0,0,1-8.94,2.52Z"
          />
          <path
            class="cv-download-banner-path"
            d="M.58,47.81,0,8.85a11.22,11.22,0,0,1,3.33-5.8A11.36,11.36,0,0,1,10.14.15L126.89,0a9.49,9.49,0,0,1,9.85,8.11q-.21,19.94-.4,39.86a11.59,11.59,0,0,0-3.51-2.63,11.73,11.73,0,0,0-5.22-1.16L8.88,43.77A9.36,9.36,0,0,0,3.19,45.2,9,9,0,0,0,.58,47.81Z"
          />
          <polygon
            class="cv-download-banner-letters"
            points="86.56 6.52 91.72 6.52 81.39 37.9 77.65 37.9 67.08 6.69 72.24 6.69 79.4 30.24 86.56 6.52"
          />
          <path
            class="cv-download-banner-letters"
            d="M61.92,15.68h5.24A10.59,10.59,0,0,0,55.92,6.19a11.17,11.17,0,0,0-10.73,8.32c-.16,2.21-.26,4.57-.25,7.08,0,3.1.16,6,.41,8.65a9.68,9.68,0,0,0,3.5,6,9.46,9.46,0,0,0,5.41,2A11.16,11.16,0,0,0,67,29.41l-5-.08a5.45,5.45,0,0,1-6.24,4.24,6.31,6.31,0,0,1-3.08-1.25,6.65,6.65,0,0,1-2.42-4.49q-.07-6.16-.16-12.32a6.2,6.2,0,0,1,2-3.08A6.27,6.27,0,0,1,55.76,11a5.57,5.57,0,0,1,4.41,1.25A5.42,5.42,0,0,1,61.92,15.68Z"
          />
        </g>
      </g>
    </svg>
    <a
      class="cv-download-arrow-circle"
      href="test"
      target="_blank"
    >
      <div
        class="cv-download-arrow"
      />
    </a>
    <p
      class="cv-download-label"
    >
      DOWNLOAD
    </p>
  </div>
</div>
`;
